opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -911,10 +930,25 @@ namespace Opm {
             WellGroupHelpers::updateGuideRates(fieldGroup, schedule(), summaryState, this->phase_usage_, reportStepIdx, simulationTime,
                                         this->wellState(), this->groupState(), comm, &this->guideRate_, pot, local_deferredLogger);
         }
-        last_report_.converged = true;
-        last_report_.assemble_time_well += perfTimer.stop();
+
+        // Maybe do a recursive call to iterate network and well controls.
+        if (network_changed) {
+            // Only re-solve network for the first nupcol newton iterations.
+            const int nupcol = schedule()[reportStepIdx].nupcol();
```

Comment:
I am still in the process in understanding the network calculation. If we want to do the iterative solution for the NUPCOL Newton iterations, you need to check the first item of the the keyword NETBALAN. By default, it only solve the nodal pressure once at the beginning of the time steps. 

And at the same time, `max_local_network_iterations` is also related to the NETBLAN keyword setup.

Reply:
Thanks for the feedback, I will look at this.